store/tikv/2pc.go

```diff
@@ -1599,6 +1600,7 @@ func (batchExe *batchExecutor) process(batches []batchMutations) error {
 					zap.Uint64("conn", batchExe.committer.connID),
 					zap.Stringer("action type", batchExe.action),
 					zap.Uint64("txnStartTS", batchExe.committer.startTS))
+				atomic.StoreUint32(&batchExe.committer.prewriteCancelled, 1)
```

Comment:
Does it work when the first error is an RPC error?

Reply:
If the first error is an RPC error, `prewriteCancelled` will be `0` at [here](https://github.com/pingcap/tidb/pull/20983/files/c177105d9f1b1adcb8756d7076ca493bdceab652). Then, an undetermined error will be set.